src/rulesets/Core/core.jl

```diff
@@ -4,5 +4,15 @@
 @non_differentiable Core.show(::Any)
 @non_differentiable Core.show(::Any, ::Any)
 
-@non_differentiable Core.apply_type(::Any, ::Any)
-@non_differentiable Core.typeof(::Any)
\ No newline at end of file
+@non_differentiable Core.apply_type(::Any...)
+@non_differentiable Core.typeof(::Any)
+
+frule((_, ẋ, _), ::typeof(typeassert), x, T) = (typeassert(x, T), ẋ)
+function rrule(::typeof(typeassert), x, T)
+    typeassert(x, T), Δ->(NoTangent(), Δ, NoTangent())
+end
+
+frule((_, _, ȧ, ḃ), ::typeof(ifelse), c, a, b) = (ifelse(c, a, b), ifelse(c, ȧ, ḃ))
+function rrule(::typeof(ifelse), c, a, b)
+    ifelse(c, a, b), Δ->(NoTangent(), NoTangent(), ifelse(c, Δ, ZeroTangent()), ifelse(c, ZeroTangent(), Δ))
```

Comment:
```suggestion
    ifelse_pullback(Δ) = (NoTangent(), NoTangent(), ifelse(c, Δ, ZeroTangent()), ifelse(c, ZeroTangent(), Δ))
    return ifelse(c, a, b), ifelse_pullback
```

Reply:
It's possible there should be another method for `rrule(::typeof(ifelse), c, a::Number, b::Number)` with `ifelse_pullback_2(Δ) = (NoTangent(), NoTangent(), ifelse(c, Δ, zero(Δ)), ifelse(c, zero(Δ), Δ))`, to be type-stable.